README.md

```diff
@@ -75,9 +75,228 @@ running Terraform.
+| **String length comparison conditions**                  |                        |                                                    |
+| `length(var.name) < 10 && length(var.name) > 0 && ...`   | `string`               | `{"minLength": 1,"maxLength": 9}`                  |
+| `length(var.name) <= 10 && length(var.name) >= 0 && ...` | `string`               | `{"minLength": 0, "maxLength": 10, }`              |
+| `length(var.name) == 5 && ...`                           | `string`               | `{"minLength": 5, "maxLength": 5"}`                |
+| **Object length comparison conditions**                  |                        |                                                    |
+| `length(var.name) < 10 && length(var.name) > 0 && ...`   | `map`, `object`        | `{"minProperties": 1,"maxProperties": 9}`          |
+| `length(var.name) <= 10 && length(var.name) >= 0 && ...` | `map`, `object`        | `{"minProperties": 0, "maxProperties": 10, }`      |
+| `length(var.name) == 5 && ...`                           | `map`, `object`        | `{"minProperties": 5, "maxProperties": 5"}`        |
+| **Array length comparison conditions**                   |                        |                                                    |
+| `length(var.name) < 10 && length(var.name) > 0 && ...`   | `list`, `tuple`, `set` | `{"minItems": 1,"maxItems": 9}`                    |
+| `length(var.name) <= 10 && length(var.name) >= 0 && ...` | `list`, `tuple`, `set` | `{"minItems": 0, "maxItems": 10, }`                |
+| `length(var.name) == 5 && ...`                           | `list`, `tuple`, `set` | `{"minItems": 5, "maxItems": 5"}`                  |
+
+### Nullable Variables
+
+If `nullable` is true in the `variable` block, then the JSON Schema will be modified to look like this. This method is primarily chosen for compatibility with react-jsonschema-form.
+
+```JSON
+"<NAME>": {
+    "anyOf": [
+        {
+            "title": "null",
+            "type": "null"
+        },
+        {
+            "title": "<TYPE>",
+            "type": "<TYPE>"
+        }
+    ],
+    "description": "<DESCRIPTION>",
+    "default": "<DEFAULT>",
+    "title": "<NAME>: Select a type"
+},
+```
+
+This is actually a slight behaviour change from the validator used by terraform. If `nullable` is unset, then terraform treats them as `nullable` by default. I chose not to implement that behaviour here and instead am making the Terraform author specify `nullable = true`. This is because otherwise schema definitions for simple programs would have to become a lot more verbose just to handle this case.
+
+### Default Handling
 
-### Nullable Variables
\ No newline at end of file
+Default handling is relatively straightforward. The default specified in Terraform is rendered to a JSON object, and added to the default field in the JSON Schema. Type checking is not performed on the default value.
```

Comment:
You could possibly mention that type checking is left out as it conforms to the standards (ie it's not a random omission).

Reply:
Yeah, was thinking about that too then forgot, I'll add that